test/unit/org/apache/cassandra/db/streaming/CassandraEntireSSTableStreamWriterTest.java

```diff
@@ -126,43 +152,98 @@ public void testBlockWriterOverWire() throws IOException
     @Test
     public void testBlockReadingAndWritingOverWire() throws Exception
     {
-        StreamSession session = setupStreamingSessionForTest();
-        InetAddressAndPort peer = FBUtilities.getBroadcastAddressAndPort();
+        testBlockReadingAndWritingOverWire(null);
+    }
+
+    /**
+     * Entire-sstable-streaming receiver will throw checksum validation failure because concurrent stats metadata
+     * update causes the actual transfered file size to be different from the one in {@link ComponentManifest}
+     */
+    @Test
+    public void testBlockReadingAndWritingOverWireWithStatsMutation() throws Exception
+    {
+        testBlockReadingAndWritingOverWire(() -> {
+            sstable.mutateAndReloadStats(stats -> stats.mutateRepairedMetadata(0, UUID.randomUUID(), false));
+            return null;
+        });
+    }
+
+    /**
+     * Entire-sstable-streaming receiver will throw checksum validation failure because concurrent index summary
+     * redistribution task will modify index summary causing actual transfered index summary size to be different from
+     * the one in {@link ComponentManifest}
+     */
+    @Test
+    public void testBlockReadingAndWritingOverWireWithIndexSummaryRedistribution() throws Exception
+    {
+        testBlockReadingAndWritingOverWire(() -> {
+
+            long nonRedistributingOffHeapSize = 0;
+            long memoryPoolBytes = 1024 * 1024;
+
+            // rewrite index summary file with new min/max index interval
+            TableMetadata origin = store.metadata();
+            MigrationManager.announceTableUpdate(origin.unbuild().minIndexInterval(1).maxIndexInterval(2).build(), true);
+
+            try (LifecycleTransaction txn = store.getTracker().tryModify(asList(sstable), OperationType.UNKNOWN))
+            {
+                IndexSummaryManager.redistributeSummaries(new IndexSummaryRedistribution(ImmutableMap.of(store.metadata().id, txn),
+                                                                                         nonRedistributingOffHeapSize,
+                                                                                         memoryPoolBytes));
+            }
+            descriptor.getMetadataSerializer().mutateRepairMetadata(descriptor, 0, UUID.randomUUID(), false);
```

Comment:
@jasonstack Do we need to mutate the repair metadata if `redistributeSummaries()` saves the index summary, which is what we seem to be testing in this case?

Reply:
good catch.. will remove it